Sources/Build/BuildPlan.swift

```diff
@@ -388,8 +388,15 @@ public final class ProductBuildDescription {
         args += ["-L", buildParameters.buildPath.asString]
         args += ["-o", binary.asString]
         args += ["-module-name", product.name]
-        args += dylibs.map({ "-l" + $0.product.name })
-
+        
+        func dylibsIn(product: ProductBuildDescription) -> [ProductBuildDescription] {
+            return product.dylibs.reduce([product], {$0 + dylibsIn(product: $1)})
+        }
+        
+        // Go through the nested dynamic libraries and add them
+        // FIXME: Use a better algorithm, because if there is a cyclic dependency an error will occur
+        args += dylibs.reduce([], {$0 + dylibsIn(product: $1)}).map({ "-l" + $0.product.name })
```

Comment:
Instead of the double reduce, which took a little bit of time to wrap my head around, could we instead have `dylibsIn(products:)` take an array of products? Feels easier for me to understand:

```swift
func dylibsIn(products: [ProductBuildDescription]) -> [ProductBuildDescription] {
    return products.reduce([], { $0 + dylibsIn(products: $1.dylibs) })
}

args += dylibsIn(in: dylibs).map({ "-l" + $0.product.name })
```

Do we get the same result?

Reply:
Applied your change and it succeeded the tests